fdbrpc/ReplicationUtils.cpp

```diff
@@ -26,6 +26,12 @@
 #include "fdbrpc/Replication.h"
 
 
+/**
+ * ratePolicy takes localitySet and ReplicationPolicy as arguments.
```

Comment:
Looking only at the function signature, I'd hope for this to answer questions like: What does it mean to rate a policy?  What is the higher level meaning of the calculation it's doing?  Over what domain is the return value?

Looking at the implementation, it looks like we return -1 if no combination of entries in localitySet satisfies the replication policy, and then (0, 1] gives an idea of how wide of a variety of replicas can be chosen that satisfy the policy, with 0 being the most diverse and 1 being the least diverse.

I'm honestly clueless on what in our codebase would use this.  Maybe a replication policy test?  I'd suggest looking at callers of ratePolicy, if you haven't already, to figure out what they're actually doing with it which should probably help fill in the useful context to provide here.